File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
  data = [1]*n
  for j in reserve:
    data[j-1] += 1 
  for i in lost:
    data[i-1] -= 1
  for i in range(n):
    if data[i] == 2:
      if i>0 and data[i-1]==0:
        data[i]=1
        data[i-1]=1
        continue
      if i<n-1 and data[i+1]==0:
        data[i]=1
        data[i+1]=1
        continue
  answer = data.count(1) + data.count(2) 
  return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.Arrays;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] student = new int[n];
        Arrays.fill(student, 1);

        for (int l : lost) {
            student[l - 1]--; 
        }

        for (int r : reserve) {
            student[r - 1]++;
        }

        for (int i = 0; i < n; i++) {
            if (student[i] == 0) {
                if (i > 0 && student[i - 1] == 2) {
                    student[i]++;
                    student[i - 1]--;
                }
                else if (i < n - 1 && student[i + 1] == 2) {
                    student[i]++;
                    student[i + 1]--;
                }
            }
        }

        int answer = 0;
        for (int s : student) {
            if (s > 0) {
                answer++;
            }
        }

        return answer;
    }
}
